### ISSUE TITLE
Add personalized defaults

### ISSUE THREAD

**Issue by grantaelliott** [deathandmayhem/jolly-roger#395]
While we try to choose reasonable defaults, there are several interface options that could reasonably have defaults determined by user preference. These would be configured from the user's profile page.

Examples:
- Chat width
- Voicechat participant visibility (collapsed / expanded)
- Puzzle list style (grouped / chronological)

**Comment by ebroder** [comment]
Is it important that these be explicitly configurable vs. just remembering the user's expressed preference? We currently store puzzle list style in `localStorage` (using `use-persisted-state`), so changes are sticky. We could pretty easily do the same for these other settings

**Comment by grantaelliott** [comment]
> Is it important that these be explicitly configurable vs. just remembering the user's expressed preference? We currently store puzzle list style in `localStorage` (using `use-persisted-state`), so changes are sticky. We could pretty easily do the same for these other settings

I had imagined changing defaults from the profile page, but I think you’re right. I’d support choosing sensible defaults and just letting settings be sticky (on a per puzzle basis, I think).

**Comment by ebroder** [comment]
Hmm. Why per-puzzle? That just seems like a lot of state to be keeping track of (that, among other things, we'll be hard pressed to garbage collect, which seems a bit annoying even if it's only a few KB)

**Comment by grantaelliott** [comment]
Actually, I think you're right. I considered it yesterday because it felt like the more natural behavior if treating adjustments as new defaults. Participation in voice chat and the real estate trade off between text chat and document are often puzzle-specific decisions, but reopening previously closed puzzles on which isn't common enough to warrant the overhead of the feature.

I am a little concerned that, with global sticky changes, the expected behavior is not obvious when multiple tabs are open. Do all open pages sync to the same settings? (Probably the more expected behavior, but potentially frustrating) If not, is the new default determined by the last adjustment made or by the state of the last page to have focus? My preference would be for chat width to be sticky and globally synced, probably in absolute value (subject to existing bounds), so that adjustments on narrow viewports don't cause wild changes.

I'd support reducing the scope of this issue to persistence of chat width but, thinking about this more deeply, I'm inclined to say our efforts are better spent elsewhere (including just choosing good defaults). I'm happy to just close this issue, unless someone feels otherwise.

Thanks for making me think it through.

**Comment by ebroder** [comment]
I think the default behavior of `use-persisted-state` would be that, yes, a change in one tab would automatically be reflected in other tabs (you can see this by opening two tabs with the puzzle list page and switching one between sorting by group and sorting by unlock). But we could conceivably arrange it other ways (e.g. all tabs write to the same `localStorage` key on edit, but only read from `localStorage` at load) by having both `use-persisted-state` and traditional `useState` variables.

**Comment by grantaelliott** [comment]
Agreed. I think that's both the more easily implemented and less unexpected behavior. It could still be surprising when applied to voice chat collapse (so I'm just dropping that as a sticky setting).

I'll keep this open for now. At some point in the future, I'll try making chat width sticky and see how well it works in practice.
